Detectors/ITSMFT/ITS/simulation/src/V3Layer.cxx

```diff
@@ -524,11 +526,21 @@ TGeoVolume *V3Layer::createModuleInnerB(Double_t xmod, Double_t ymod, Double_t z
 {
   Double_t xtot, ytot, ztot, zchip;
   Double_t xpos, ypos, zpos;
-  char volumeName[30];
+  Bool_t dummyChip;
+  char chipName[30], sensName[30], volumeName[30];
+
+  // For material budget studies
+  if (mBuildLevel < 6)
+    dummyChip = kFALSE;  // will be made of Si
+  else
+    dummyChip = kTRUE;   // will be made of Air
 
   // First create the single chip
-  zchip = zmod / sIBChipsPerRow;
-  TGeoVolume *chipVol = createChip(xmod, ymod, zchip);
+  snprintf(chipName, 30, "%s%d", GeometryTGeo::getITSChipPattern(), mLayerNumber);
+  snprintf(sensName, 30, "%s%d", GeometryTGeo::getITSSensorPattern(), mLayerNumber);
```

Comment:
better use `std::string` operations, or make a constant variable for the `char[]` size, use it in the definitions and here. In principle, only max 29 characters can be copied here and a terminating `0` must be set at the end. No worries with `std::string` though